app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -16,7 +16,25 @@ def registration_args
   end
 
   def insecure
-    registration_params['insecure'] ? '--insecure' : ''
+    return '' unless registration_params['insecure']
+
+    if download_utility == 'curl'
```

Comment:
What do you think to add something like this:
`
utility = {
 curl => { insecure => '--insecure', download => 'curl -sS },
wget => { insecure => '--no-check-certificate, download => 'wget --no-verbose -O -' }
}
`

This can be extended to other tools later on.